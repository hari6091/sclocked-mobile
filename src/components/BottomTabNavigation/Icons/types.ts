import { SvgProps } from 'react-native-svg';

export interface IconProps extends SvgProps {
  focused: boolean;
  color?: string;
  size?: number;
}
