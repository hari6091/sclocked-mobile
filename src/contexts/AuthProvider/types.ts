interface ILogin {
  email: string;
  password: string;
}

export interface IUser {
  email?: string;
  username?: string;
  token?: string;
}

export interface IContext extends IUser {
  signIn: ({ email, password }: ILogin) => Promise<void>;
  logout: () => void;
}

export interface IAuthProvider {
  children: JSX.Element;
}
