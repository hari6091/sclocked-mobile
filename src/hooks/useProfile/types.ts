export interface MyUser {
  uuid?: string;
  id?: string;
  name: string;
  tags: string;
  matricula: string;
  email: string;
  disciplinaOUcargo: string;
  role: string;
  createdAt: string;
}

export interface MyUserSalas {
  uuid?: string;
  id?: string;
  name: string;
  numero: string;
  grupo: string;
  status: string;
  users: MyUser[];
}

export type IUserUpdate = {
  id?: string | undefined;
  name: string | undefined;
  tags: string | undefined;
  matricula: string | undefined;
  disciplinaOUcargo: string | undefined;
  email: string | undefined;
  password: string;
  confPassword: string;
  role: string | undefined;
};
