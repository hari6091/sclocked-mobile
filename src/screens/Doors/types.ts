import { screens } from "../../constants";
import { MainTabScreenProps } from "../../routes/types";

export type DoorsScreenProps = MainTabScreenProps<screens.DOOR>;

export interface ISalaOpt {
  salaNumber: string;
  status: string;
}
