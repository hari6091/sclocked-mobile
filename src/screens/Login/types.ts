import { screens } from "../../constants";
import { RootStackScreenProps } from "../../routes/types";

export type LoginScreenProps = RootStackScreenProps<screens.LOGIN>;

export type SignInType = {
  email: string;
  password: string;
};
