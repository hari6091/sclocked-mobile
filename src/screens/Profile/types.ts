import { screens } from "../../constants";
import { MainTabScreenProps } from "../../routes/types";

export type ProfileScreenProps = MainTabScreenProps<screens.PROFILE>;
