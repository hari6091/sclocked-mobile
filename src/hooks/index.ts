export * from "./useAuth";
export * from "./useProfile";
export * from "./useSalaAuth";
