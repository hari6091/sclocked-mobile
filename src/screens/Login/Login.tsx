import {
  Center,
  Text,
  Button,
  HStack,
  VStack,
  FormControl,
  Input,
  Icon,
} from "native-base";
import React, { useState } from "react";
import * as Yup from "yup";
import { FontAwesome, MaterialIcons } from "@expo/vector-icons";
import { LoginScreenProps } from "./types";
import { useAuth } from "../../hooks";
import { useFormik } from "formik";

const SignInSchema = () =>
  Yup.object().shape({
    email: Yup.string()
      .email("Insira um email válido!")
      .required("Este campo é obrigatório!"),
    password: Yup.string().required("Este campo é obrigatório!"),
  });

const Login = ({ navigation }: LoginScreenProps) => {
  const { signIn } = useAuth();
  const [visibility, setVisibility] = useState<boolean>(false);

  const {
    values,
    touched,
    errors,
    handleChange,
    handleBlur,
    handleSubmit,
    isSubmitting,
  } = useFormik({
    initialValues: { email: "", password: "" },
    validationSchema: SignInSchema,
    onSubmit: () => {
      signIn(values);
    },
  });

  const getErrorMsg = (field: keyof typeof values) => {
    return touched[field] ? errors[field] : undefined;
  };

  return (
    <>
      <HStack
        w="100%"
        h="20%"
        justifyContent="center"
        alignItems="center"
        borderBottomRadius="18px"
        bg="#28374D"
        safeArea
      >
        <Icon as={FontAwesome} name="lock" size="12" color="#2486CE" />
        <Text
          fontSize="36px"
          textAlign="center"
          color="#EBE8E1"
          fontWeight={700}
          testID="app-title"
        >
          Slocked
        </Text>
      </HStack>
      <Center flex={1} flexDirection="column" px="10" bg="#FFFFFF">
        <VStack justifyContent="center" w="100%">
          <Center p="14px">
            <FormControl isInvalid={!!getErrorMsg("email")} mb="6">
              <FormControl.Label testID="email-label">
                Digite seu email:
              </FormControl.Label>
              <Input
                placeholder="Email"
                borderColor="#2486CE"
                borderRadius="16px"
                h="64px"
                fontSize="16px"
                onChangeText={handleChange("email")}
                onBlur={handleBlur("email")}
                testID="email-input"
              />
              <FormControl.ErrorMessage testID="email-error">
                {getErrorMsg("email")}
              </FormControl.ErrorMessage>
            </FormControl>
            <FormControl mb="12" isInvalid={!!getErrorMsg("password")}>
              <FormControl.Label testID="password-label">
                Digite sua senha:
              </FormControl.Label>
              <Input
                placeholder="Senha"
                type={visibility ? "text" : "password"}
                borderColor="#2486CE"
                borderRadius="16px"
                h="64px"
                fontSize="16px"
                onChangeText={handleChange("password")}
                onBlur={handleBlur("password")}
                testID="password-input"
                rightElement={
                  <Icon
                    as={MaterialIcons}
                    name={visibility ? "visibility" : "visibility-off"}
                    color="#2486CE"
                    size="24px"
                    mr="16px"
                    onPress={() => setVisibility(!visibility)}
                  />
                }
              />
              <FormControl.ErrorMessage testID="password-error">
                {getErrorMsg("password")}
              </FormControl.ErrorMessage>
            </FormControl>
            <VStack w="full" space="6">
              <Button
                h="64px"
                w="full"
                borderRadius="30px"
                bg="#2486CE"
                _text={{ fontWeight: 700, fontSize: "22px" }}
                onPress={() => handleSubmit()}
                isDisabled={isSubmitting}
                isLoading={isSubmitting}
                testID="login-button"
              >
                Login
              </Button>
            </VStack>
          </Center>
        </VStack>
      </Center>
    </>
  );
};

export default Login;
