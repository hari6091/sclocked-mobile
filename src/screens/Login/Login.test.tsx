import React from "react";
import Login from "./Login";
import { act, fireEvent, render } from "../../test/test-utils";
import { LoginScreenProps } from "./types";
import * as auth from "../../hooks/useAuth";

jest.mock("../../hooks/useAuth", () => ({
  __esModule: true,
  useAuth: jest.fn(() => ({
    signIn: jest.fn(),
    logout: jest.fn(),
  })),
}));

describe("Login screen", () => {
  const navigation = {
    setOptions: jest.fn(),
  } as unknown as LoginScreenProps["navigation"];

  const route = {
    params: undefined,
  } as unknown as LoginScreenProps["route"];

  const signInMocked = jest.fn();
  const logoutMocked = jest.fn();

  beforeEach(() => {
    jest.spyOn(auth, "useAuth").mockReturnValue({
      signIn: signInMocked,
      logout: logoutMocked,
    });
  });

  afterEach(() => {
    signInMocked.mockReset();
  });

  it("renders screen, and email and password fields correctly", () => {
    const { getByTestId, toJSON } = render(
      <Login navigation={navigation} route={route} />
    );

    expect(getByTestId("email-label")).toBeTruthy();
    expect(getByTestId("email-input")).toBeTruthy();

    expect(getByTestId("password-label")).toBeTruthy();
    expect(getByTestId("password-input")).toBeTruthy();

    expect(toJSON()).toMatchSnapshot();
  });

  it("should prevent sign in if it has an invalid field", async () => {
    const { getByTestId } = render(
      <Login navigation={navigation} route={route} />
    );

    await act(async () => {
      fireEvent.press(getByTestId("login-button"));
    });

    expect(signInMocked).toHaveBeenCalledTimes(0);
  });

  it("should sign in", async () => {
    const { getByTestId, getByText } = render(
      <Login navigation={navigation} route={route} />
    );

    fireEvent.changeText(getByTestId("email-input"), "admin@admin.com");
    fireEvent.changeText(getByTestId("password-input"), "qwe123");

    await act(async () => {
      fireEvent.press(getByTestId("login-button"));
    });

    expect(signInMocked).toHaveBeenCalledTimes(1);
    expect(signInMocked).toHaveBeenNthCalledWith(1, {
      email: "admin@admin.com",
      password: "qwe123",
    });
  });
});
