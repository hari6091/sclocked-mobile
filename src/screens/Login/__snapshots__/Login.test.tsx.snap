// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login screen renders screen, and email and password fields correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={{}}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#28374D",
        "borderBottomLeftRadius": 18,
        "borderBottomRightRadius": 18,
        "flexDirection": "row",
        "height": "20%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Text />
    <View
      dataSet={{}}
      style={{}}
    />
    <Text
      dataSet={{}}
      style={
        {
          "backgroundColor": undefined,
          "color": "#EBE8E1",
          "fontFamily": undefined,
          "fontSize": 36,
          "fontStyle": "normal",
          "fontWeight": "700",
          "letterSpacing": 0,
          "lineHeight": 54,
          "textAlign": "center",
          "textDecorationLine": undefined,
        }
      }
      testID="app-title"
    >
      Slocked
    </Text>
  </View>
  <View
    dataSet={{}}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "display": "flex",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "paddingLeft": 40,
        "paddingRight": 40,
      }
    }
  >
    <View
      dataSet={{}}
      style={
        {
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        dataSet={{}}
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "center",
            "paddingBottom": 14,
            "paddingLeft": 14,
            "paddingRight": 14,
            "paddingTop": 14,
          }
        }
      >
        <View
          dataSet={{}}
          isInvalid={false}
          style={
            {
              "marginBottom": 24,
              "width": "100%",
            }
          }
        >
          <View
            dataSet={{}}
            feedbackId="field-3-feedback"
            hasFeedbackText={true}
            hasHelpText={false}
            helpTextId="field-3-helptext"
            isDisabled={false}
            isInvalid={false}
            isReadOnly={false}
            isRequired={false}
            labelId="field-3-label"
            nativeID="field-3-label"
            setHasFeedbackText={[Function]}
            setHasHelpText={[Function]}
            style={
              {
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "marginBottom": 4,
                "marginTop": 4,
              }
            }
            testID="email-label"
          >
            <Text
              dataSet={{}}
              style={
                {
                  "backgroundColor": undefined,
                  "color": "#737373",
                  "fontFamily": undefined,
                  "fontSize": 14,
                  "fontStyle": "normal",
                  "fontWeight": "500",
                  "letterSpacing": 0,
                  "lineHeight": 21,
                  "textDecorationLine": undefined,
                }
              }
            >
              Digite seu email:
            </Text>
          </View>
          <View
            dataSet={{}}
            isFocused={false}
            style={
              {
                "alignItems": "center",
                "borderColor": "#2486CE",
                "borderRadius": 16,
                "borderWidth": 1,
                "flexDirection": "row",
                "height": 64,
                "overflow": "hidden",
              }
            }
          >
            <TextInput
              accessibilityDescribedBy="field-3-feedback"
              accessible={true}
              dataSet={{}}
              disabled={false}
              editable={true}
              isRequired={false}
              nativeID="field-3-input"
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onKeyPress={[Function]}
              placeholder="Email"
              placeholderTextColor="#a3a3a3"
              readOnly={false}
              required={false}
              secureTextEntry={false}
              style={
                {
                  "backgroundColor": "transparent",
                  "color": "#171717",
                  "flex": 1,
                  "fontFamily": undefined,
                  "fontSize": 16,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                  "height": "100%",
                  "paddingBottom": 8,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "paddingTop": 8,
                  "width": "100%",
                }
              }
              testID="email-input"
            />
          </View>
        </View>
        <View
          dataSet={{}}
          isInvalid={false}
          style={
            {
              "marginBottom": 48,
              "width": "100%",
            }
          }
        >
          <View
            dataSet={{}}
            feedbackId="field-4-feedback"
            hasFeedbackText={true}
            hasHelpText={false}
            helpTextId="field-4-helptext"
            isDisabled={false}
            isInvalid={false}
            isReadOnly={false}
            isRequired={false}
            labelId="field-4-label"
            nativeID="field-4-label"
            setHasFeedbackText={[Function]}
            setHasHelpText={[Function]}
            style={
              {
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "marginBottom": 4,
                "marginTop": 4,
              }
            }
            testID="password-label"
          >
            <Text
              dataSet={{}}
              style={
                {
                  "backgroundColor": undefined,
                  "color": "#737373",
                  "fontFamily": undefined,
                  "fontSize": 14,
                  "fontStyle": "normal",
                  "fontWeight": "500",
                  "letterSpacing": 0,
                  "lineHeight": 21,
                  "textDecorationLine": undefined,
                }
              }
            >
              Digite sua senha:
            </Text>
          </View>
          <View
            dataSet={{}}
            isFocused={false}
            style={
              {
                "alignItems": "center",
                "borderColor": "#2486CE",
                "borderRadius": 16,
                "borderWidth": 1,
                "flexDirection": "row",
                "height": 64,
                "overflow": "hidden",
              }
            }
          >
            <TextInput
              accessibilityDescribedBy="field-4-feedback"
              accessible={true}
              dataSet={{}}
              disabled={false}
              editable={true}
              isRequired={false}
              nativeID="field-4-input"
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onKeyPress={[Function]}
              placeholder="Senha"
              placeholderTextColor="#a3a3a3"
              readOnly={false}
              required={false}
              secureTextEntry={true}
              style={
                {
                  "backgroundColor": "transparent",
                  "color": "#171717",
                  "flex": 1,
                  "fontFamily": undefined,
                  "fontSize": 16,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                  "height": "100%",
                  "paddingBottom": 8,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "paddingTop": 8,
                  "width": "100%",
                }
              }
              testID="password-input"
            />
            <View
              dataSet={{}}
              style={{}}
            />
            <Text />
          </View>
        </View>
        <View
          dataSet={{}}
          style={
            {
              "width": "100%",
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            colorScheme="primary"
            dataSet={{}}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#2486CE",
                "borderRadius": 30,
                "flexDirection": "row",
                "height": 64,
                "justifyContent": "center",
                "paddingBottom": 10,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 10,
                "width": "100%",
              }
            }
            testID="login-button"
          >
            <View
              dataSet={{}}
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
              test={true}
            >
              <View
                dataSet={{}}
                style={{}}
              >
                <Text
                  dataSet={{}}
                  style={
                    {
                      "backgroundColor": undefined,
                      "color": "#fafafa",
                      "fontFamily": undefined,
                      "fontSize": 22,
                      "fontStyle": "normal",
                      "fontWeight": "700",
                      "letterSpacing": 0,
                      "lineHeight": 33,
                      "textDecorationLine": undefined,
                    }
                  }
                >
                  Login
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
